### PR TITLE
BUG: Respect check_dtype in assert_series_equal

### LINKED ISSUE #32747: BUG: Regression in assert_frame_equal check_dtype for extension dtypes
Consider this small example of two DataFrames, one with an Int64 extension dtype, the other with the same values but object dtype:

```
df1 = pd.DataFrame({'a': pd.array([1, 2, 3], dtype="Int64")}) 
df2 = df1.astype(object)   
```

With pandas 1.0.1, this passes `assert_frame_equal` with the `check_dtype=False`:

```
In [5]: pd.testing.assert_frame_equal(df1, df2)  
...
Attribute "dtype" are different
[left]:  Int64
[right]: object

In [6]: pd.testing.assert_frame_equal(df1, df2, check_dtype=False)  
```

but with master (since https://github.com/pandas-dev/pandas/pull/32570, see my comment there, cc @jbrockmendel), this fails:

```
In [2]: pd.testing.assert_frame_equal(df1, df2, check_dtype=False)   
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-2-d2d792202db1> in <module>
----> 1 pd.testing.assert_frame_equal(df1, df2, check_dtype=False)

~/scipy/pandas/pandas/_testing.py in assert_frame_equal(left, right, check_dtype, check_index_type, check_column_type, check_frame_type, check_less_precise, check_names, by_blocks, check_exact, check_datetimelike_compat, check_categorical, check_like, obj)
   1378                 check_datetimelike_compat=check_datetimelike_compat,
   1379                 check_categorical=check_categorical,
-> 1380                 obj=f'{obj}.iloc[:, {i}] (column name="{col}")',
   1381             )
   1382 

~/scipy/pandas/pandas/_testing.py in assert_series_equal(left, right, check_dtype, check_index_type, check_series_type, check_less_precise, check_names, check_exact, check_datetimelike_compat, check_categorical, check_category_order, obj)
   1177         )
   1178     elif is_extension_array_dtype(left.dtype) or is_extension_array_dtype(right.dtype):
-> 1179         assert_extension_array_equal(left._values, right._values)
   1180     elif needs_i8_conversion(left.dtype) or needs_i8_conversion(right.dtype):
   1181         # DatetimeArray or TimedeltaArray

~/scipy/pandas/pandas/_testing.py in assert_extension_array_equal(left, right, check_dtype, check_less_precise, check_exact)
   1017     """
   1018     assert isinstance(left, ExtensionArray), "left is not an ExtensionArray"
-> 1019     assert isinstance(right, ExtensionArray), "right is not an ExtensionArray"
   1020     if check_dtype:
   1021         assert_attr_equal("dtype", left, right, obj="ExtensionArray")

AssertionError: right is not an ExtensionArray
```

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes #32747
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

I'm not sure if https://github.com/pandas-dev/pandas/blob/master/pandas/_testing.py#L1174 needs to change too?

cc @jorisvandenbossche

### CODE REVIEW

File: pandas/tests/util/test_assert_series_equal.py [hunk 1]
```diff
@@ -194,3 +195,39 @@ def test_series_equal_categorical_mismatch(check_categorical):
...
+
+Attribute "dtype" are different
+\\[left\\]:  Int64
+\\[right\\]: int64"""
+
+    if check_dtype:
+        with pytest.raises(AssertionError, match=msg):
+            tm.assert_series_equal(left, right, check_dtype=check_dtype)
+    else:
+        tm.assert_series_equal(left, right, check_dtype=check_dtype)
```
Comment (jorisvandenbossche) [inline pandas/tests/util/test_assert_series_equal.py]: Can you just test both True / False cases inline here without the parametrization? (it actually gets you rid of the if/else, while for the rest not adding any code)

Comment (jorisvandenbossche) [reply pandas/tests/util/test_assert_series_equal.py]: (and same below)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for looking into this!

Can you also add a test for the `assert_frame_equal` case? (so to ensure it is properly passed through)

**Comment by jreback** [comment]
can you add a whatsnew note (can put in bug fixes other section)

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -401,6 +401,7 @@ Other
 - Bug in :meth:`DataFrame.to_records` incorrectly losing timezone information in timezone-aware ``datetime64`` columns (:issue:`32535`)
 - Fixed :func:`pandas.testing.assert_series_equal` to correctly raise if left object is a different subclass with ``check_series_type=True`` (:issue:`32670`).
 - :meth:`IntegerArray.astype` now supports ``datetime64`` dtype (:issue:32538`)
+- Fixed bug in :func:`pandas._testing.assert_series_equal` where dtypes were checked for ``Interval`` and ``ExtensionArray`` operands when ``check_dtype`` was ``False`` (:issue:`32747`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.1.0.rst]:
```suggestion
- Fixed bug in :func:`pandas.testing.assert_series_equal` where dtypes were checked for ``Interval`` and ``ExtensionArray`` operands when ``check_dtype`` was ``False`` (:issue:`32747`)
```

(that's the public API)

**Comment by simonjayhawkins** [comment]
> can you add a whatsnew note (can put in bug fixes other section)

IIUC the bug is only on master, not released. Is the whatsnew needed?

**Comment by jorisvandenbossche** [comment]
> IIUC the bug is only on master, not released. Is the whatsnew needed?

Part of the fix did fix something compared to 1.0 (eg for intervals)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @dsaxton !

**Comment by simonjayhawkins** [comment]
> Part of the fix did fix something compared to 1.0 (eg for intervals)

If this fixes something that worked in 1.0, then should this be backported?

**Comment by jorisvandenbossche** [comment]
Sorry, I meant: it's partly fixing a bug that is in 1.0 (hence the whatsnew note is fine), and partly a regression that is only in master (which doesn't need a whatsnew). 
But the bug from 1.0 is not a regression nor critical, so shouldn't need a backport

**Comment by simonjayhawkins** [comment]
ah. my bad.

**Pull request merged**